src/ipld/util.rs

```diff
@@ -550,3 +554,11 @@ impl<DB: Blockstore + Send + Sync + 'static, T: Iterator<Item = Tipset> + Unpin>
         }
     }
 }
+
+fn ipld_to_cid(ipld: Ipld) -> Option<Cid> {
```

Comment:
nit: Could we move this to e.g. `src/ipld/util.rs`? This snippet is duplicated in several places in the codebase.